libs/contact/src/effects/contact.effects.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable, Inject } from '@angular/core';
+import { DaffContactDriver } from '../driver/injection-tokens/contact-driver.token';
+import { DaffContactServiceInterface } from '../driver/interfaces/contact-service.interface';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { Action } from '@ngrx/store';
+import { Observable, of } from 'rxjs';
+import { DaffContactActionTypes, DaffContactSubmit, DaffContactCancel, DaffContactSuccessSubmit, DaffContactFailedSubmit, DaffContactRetry } from '../actions/contact.actions';
+import { switchMap, map, catchError } from 'rxjs/operators';
+
+@Injectable()
+export class DaffContactEffects<T, V>{
+  constructor(
+    private actions$: Actions,
+    @Inject(DaffContactDriver) private driver: DaffContactServiceInterface<T, V>
+  ) { }
+
+  trySubmission$: Observable<Action> = createEffect(() => this.actions$.pipe(
+    ofType(DaffContactActionTypes.ContactSubmitAction,
+      DaffContactActionTypes.ContactRetryAction,
+      DaffContactActionTypes.ContactCancelAction),
+    switchMap((action: DaffContactSubmit<T> | DaffContactRetry<T> | DaffContactCancel) => {
+      if ((action.type === DaffContactActionTypes.ContactCancelAction)) {
+        return of(action);
+      }
+      else if (action instanceof DaffContactSubmit || action instanceof DaffContactRetry) {
+        return this.driver.send(action.payload).pipe(
+          map((resp: V) => {
+            return new DaffContactSuccessSubmit();
+          }),
+          catchError(error => {
+            return of(new DaffContactFailedSubmit(['Failed to submit']));
+          })
+        )
+      }
+    }),
+    ofType(DaffContactActionTypes.ContactSuccessSubmitAction, DaffContactActionTypes.ContactFailedSubmitAction)
```

Comment:
What is this for? Is it needed?

Reply:
I'm now doubting myself on this, but I believe it filters the effect to only return either a `SuccessSubscribe` and a `FailedSubmit` observable. This really only matters so the `DaffContactCancel` is submitted it returns empty